presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -252,6 +252,9 @@ default ConnectorTableHandle beginDelete(ConnectorSession session, ConnectorTabl
         throw new PrestoException(NOT_SUPPORTED, "This connector does not support deletes");
     }
 
+    default void beginSelect(ConnectorSession session, ConnectorTableHandle tableHandle, Optional<ConnectorTableLayoutHandle> layoutHandle, Collection<ColumnHandle> columnHandles)
```

Comment:
This is a direct translation from `TableScanNode` which has `TableLayoutHandle` optional. I rely on `TableScanNode` in this mechanism, unfortunately.